packages/legend-application/src/components/shared/PackageableElementOptionRenderer.tsx

```diff
@@ -14,23 +14,64 @@
  * limitations under the License.
  */
 
-import type { PackageableElement } from '@finos/legend-graph';
+import type {
+  PackageableElement,
+  GraphManagerState,
+} from '@finos/legend-graph';
 import type { PackageableElementOption } from '../../stores/shared/PackageableElementOption.js';
 
-export const getPackageableElementOptionalFormatter = (props?: {
+export const getPackageableElementOptionFormatter = (props: {
   darkMode?: boolean;
+  graphManagerState: GraphManagerState;
 }): ((
   option: PackageableElementOption<PackageableElement>,
 ) => React.ReactNode) =>
   function PackageableElementOptionLabel(
     option: PackageableElementOption<PackageableElement>,
   ): React.ReactNode {
-    const className = props?.darkMode
+    const className = props.darkMode
       ? 'packageable-element-format-option-label--dark'
       : 'packageable-element-format-option-label';
 
+    const getColor = (element: PackageableElement): string => {
+      switch (true) {
+        case props.graphManagerState.isSystemElement(element):
+          return 'system';
+        case props.graphManagerState.isGeneratedElement(element):
+          return 'generated';
+        case props.graphManagerState.isMainGraphElement(element):
+          return 'main';
+        case props.graphManagerState.isDependencyElement(element):
+          return 'dependency';
+        default:
+          return '';
+      }
+    };
+
+    const getToolTip = (element: PackageableElement): string => {
+      switch (true) {
+        case props.graphManagerState.isSystemElement(element):
+          return 'system';
+        case props.graphManagerState.isGeneratedElement(element):
+          return 'generated';
+        case props.graphManagerState.isMainGraphElement(element):
+          return '';
+        case props.graphManagerState.isDependencyElement(element):
+          return 'dependency';
+        default:
+          return '';
+      }
+    };
+
+    const optionColor = getColor(option.value);
+    const toolTip = getToolTip(option.value);
+
     return (
       <div className={className}>
+        <div
+          title={`${toolTip} element`}
```

Comment:
when `title=undefined`, nothing will be shown, so that's why we should have `tooltipText = undefined` for `main graph` case

Reply:
! very true makes sense 👍